src/mysql_conn.erl

```diff
@@ -477,13 +478,26 @@ handle_call(rollback, {FromPid, _},
                                                ?cmd_timeout),
     setopts(SockMod, Socket, [{active, once}]),
     State1 = update_state(Res, State),
-    {reply, ok, State1#state{transaction_levels = L}};
+    {reply, ok, State1#state{transaction_levels = L1}};
+handle_call(rollback, {FromPid, _},
+            State = #state{status = Status, transaction_levels = TransLevels})
+  when Status band ?SERVER_STATUS_IN_TRANS /= 0 ->
+    % This case happens when we start transaction a and then transaction b
```

Comment:
Just a style nit: Use `%%` for comments aligned with the rest of the code. Use `%` for end-of-line comments. This is by the [Emacs mode](https://www.erlang.org/doc/man/erlang.el.html#indent) (which is the official Erlang style):

> * Lines with one %-character is indented to the right of the code. The column is specified by the variable comment-column, by default column 48 is used.
> * Lines with two %-characters will be indented to the same depth as code would have been in the same situation.
> * Lines with three of more %-characters are indented to the left margin.